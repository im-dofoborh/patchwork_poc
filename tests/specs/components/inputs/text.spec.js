
import { mount, createLocalVue } from '@vue/test-utils';
import TextInput from '~/components/inputs/Text';

describe('TextInput component', () => {
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('text-input', TextInput);
    wrapper = mount(TextInput, {
      localVue
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits input event on input', () => {
    const input = wrapper.find('input');
    const value = 'foo';
    input.element.value = value;
    input.trigger('input');
    expect(wrapper.emitted().input[0]).toEqual([value]);
  });
});
