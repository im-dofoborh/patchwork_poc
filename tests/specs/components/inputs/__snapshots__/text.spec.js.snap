// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component renders default variant 1`] = `<input class="input-text">`;
